### DISCUSSION TITLE
New Alien Race (Grasslings)

### DISCUSSION THREAD

**Discussion by lumbar527** [endless-sky/endless-sky#8806] [Ideas]
I'm currently working on new alien races. What do you think about my current details?
[Brakak.txt](https://github.com/endless-sky/endless-sky/files/11627897/Brakak.txt)
[Grasslings.txt](https://github.com/endless-sky/endless-sky/files/11627898/Grasslings.txt)

## Status
One semi-complete ship and one incomplete ship. I need to either get help or learn more blender to take the semi- away.
Anyone have ideas or suggestions?

[Will be closed if I start doing stuff on Discord]

**Comment by lumbar527** [comment]
Here's a [mostly | completely] finished ship:
![grassling tree I-safari scale](https://github.com/endless-sky/endless-sky/assets/125756761/bbe56d75-57a7-46cb-b44d-ede53bbd227d)
I should probably add more "grass".

**Reply by lumbar527** [reply]
Slightly better:
![grassling tree II](https://github.com/endless-sky/endless-sky/assets/125756761/93fb73a5-77aa-41c4-b436-eef6361fa3e3)

**Comment by mOctave** [comment]
It's unclear whether you intend this to be for vanilla or a plugin.

If this is for a plugin, it's great! New plugin races are definitely a good thing to have, and I look forward to potentially playing it at some point.

If it's for vanilla, it's still a really neat idea, but keep in mind that:
1. Vanilla races usually take at least a year to develop, to the best of my knowledge.
1. Lore is a big part of a faction. It's all well and good to have the "Grasslings," but you'll need to do a lot of justification and the final result might not be exactly what you had in mind.
1. The galaxy is sort of running out of space at this point, so you should avoid too many connections with other factions.
1. This is still well below the art quality expected of vanilla content. While there's nothing wrong with that (check out some of MZ's earlier work), it will likely take you months of Blendering to get to the point where this meets the high bar of current acceptability in modelling. If this is something you plan to pursue yourself, I would highly recommend doing some ship splices of existing ES ships to get a feel for the art style and discover some of the tools used therein.

**Reply by lumbar527** [reply]
I'm not sure if I want vanilla or plugin. Vanilla would be awesome, but a plugin would be much easier.
Yeah, I'm not sure how to get that high detail on ships. (I'm also not sure where. Maybe in the core?)

**Comment by lumbar527** [comment]
Here's my current blend files for the ship: [blend.zip](https://github.com/endless-sky/endless-sky/files/11599947/blend.zip)
If you want to help, feel free to edit it.

**Comment by lumbar527** [comment]
One ship is the same as before, the other doesn't even have all its "grass".
[tree/bush.zip](https://github.com/endless-sky/endless-sky/files/11600231/tree.bush.zip)

**Comment by lumbar527** [comment]
What do you think about the Pug being part of this? (In the document at the top)
I would also like some feedback from major people, if they have time.

**Reply by DarcyManoel** [reply]
The Pug should be the last resort for a plot device. Stories should maintain regional or tangentially related situations as the crux of a plot, we really don't want to make the Pug get involved in every affair within the galaxy.

**Reply by lumbar527** [reply]
OK, then I should make a second race?

**Comment by lumbar527** [comment]
The second race:
[Brakak.txt](https://github.com/endless-sky/endless-sky/files/11620073/Brakak.txt)
![brakak cudgel](https://github.com/endless-sky/endless-sky/assets/125756761/11fd6695-ddef-4adb-a7d8-f44269445cec)
The Cudgel, a basic ship. (With a simple design)

**Comment by lumbar527** [comment]
Where do you think the systems should be? (Screenshot taken through quyykk's plugin editor, with files switched for newest map. Colors missing because of labeling a color like `republic color`, which feature has not been added to the editor.)
<img width="1680" alt="grassling map figure" src="https://github.com/endless-sky/endless-sky/assets/125756761/67603bb4-3cd5-45cf-95a9-f9d5750306c1">
Circled places are places where there are dots that could be systems. An alternative would be to add some other galaxy or something.

**Reply by Saugia** [reply]
There are (vanilla) content creators working in all of those places already except above Gegno space, which I'm somewhat skeptical of myself. While that doesn't mean they're simply taken and you can't use those areas, you should communicate with them first. The top left however is planned for some core content someday so isn't available.

To be honest, I don't see this Grassling race appropriate for vanilla. It is quite out of the blue for a discussion, has no prior lore set up, in which those working on some races have had a lot of back and forth with other content creators, devs, etc.. for many months to years. You seem very new to the game still and I'd like to see more game knowledge and practice with both lore and models before you consider working towards vanilla. I strongly suggest you work with the Grasslings as a plug-in. You wanting to work on ships, visuals, systems, etc. prior to any developed lore is something a ton of people want to do for this sort of thing, but the cooperation and lore required for vanilla is more important than any of these aspects. The lore should be before anything else, before thinking about ships, before map locations, etc.. 

After looking at the text provided, it is some groundwork, but is very bare-bones. New vanilla races should be adding something new and unique to the game, which can be hard to do and requires a lot of work. So far, the Grasslings add "protective species with different ship things that are wary outsiders", which is common among new races. Also, simply giving them a relation to another, such as the Pug, takes a lot more setup than just declaring it. The timeline is okay to have, but is again very bare-bones.

If you're looking to add your own representation of your own work to the game, I again strongly suggest a plug-in. If you're looking to contribute to vanilla with another race (which will probably be a very long time from now, considering the others being worked on), I suggest practicing with plug-ins and making ships, lore/engaging in lore discussions with other creators, etc.. before returning to attempt a vanilla race.

**Reply by Saugia** [reply]
I read up and saw that you aren't sure if you'd like it for vanilla or a plug-in and a plug-in would be a lot easier. In that case, a plug-in is definitely something I recommend, and I also recommend discussing work on this in the #plugin-chat channel on discord, as there are a lot of great plug-in creators there that can offer advice and help.

With a plug-in, you're free to do as you please, you can make all the ships and systems, and we have a great dedicated member base who really enjoy extra plug-in content. That's one of the main things our creator, MZ, wanted to come from the game.

**Reply by lumbar527** [reply]
I think I'll at least begin with a plugin. Maybe even after a million years of being a plugin, then another million, I could contribute it to Endless Sky.

**Comment by lumbar527** [comment]
Closing for lore development and above replies: https://github.com/endless-sky/endless-sky/discussions/8806#discussioncomment-6062286.
